dsa-23au/java-dsa/pswish-natmcl/pswish-app/src/main/java/com/pswishcorp/app/Driver.java

```diff
@@ -1,7 +1,51 @@
 package com.pswishcorp.app;
+import java.util.HashMap;
+import java.util.Map;
 
 public class Driver {
     int distanceFromHome;
-    String caryType;
-    
-}
+    String driverName;
+    String carMake;
+    String carModel;
+    String home;
+    String school;
+
+   public Driver(int distanceFromHome,  String driverName,
+            String carMake, String carModel, String home, String school) {
+        this.distanceFromHome = distanceFromHome;
+        this.driverName = driverName;
+        this.carMake = carMake;
+        this.carModel = carModel;
+        this.home = home;
+        this.school = school;
+       }
+public static void main(String[] args) {
+        Driver driver1 = new Driver(11, "Paul", "Audi", "Q5", "Millersylvania", "Evergreen");
+        Driver driver2 = new Driver(15, "Nathan", "Hyndai", "Genesis", "Lakewood", "Evergreen");
+        Driver driver3 = new Driver(86, "Paul", "Audi", "Q5", "Gardiner", "Evergreen");
+        }
+
+    // Remove the throws exception when finished then check for all following tests
+    public Object getDistanceFromHome() throws Exception {
+        throw new Exception("Not yet implemented, v0.4");
+    }
+
+    public Object getDriverName(String driver) {
+        Map<String, String> drivers = new HashMap<>();
+        return drivers;
+    }
+
+    public Object getCarMake() throws Exception {
```

Comment:
what does throws exception mean? I never covered "throws" keyword and exception handling (I think it's called that)

Reply:
'throws exception' causes the compile to spit out an error when compiling and is meant to be a placeholder and a reminder to update the code. Paul mentioned in class to change it to a runtime exception.

Reply:
Here is some documentation: https://docs.oracle.com/javase/tutorial/essential/exceptions/throwing.html but think of it like you can use it as reminder to fix it later that should let you compile. My way isn’t the best way so I am looking to make changes soon.